src/integration-test/java/io/aiven/kafka/connect/KafkaIntegrationBase.java

```diff
@@ -0,0 +1,99 @@
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.Future;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+import org.testcontainers.containers.KafkaContainer;
+
+
+public interface KafkaIntegrationBase {
+
+    default AdminClient newAdminClient(final KafkaContainer kafka) {
+        final Properties adminClientConfig = new Properties();
+        adminClientConfig.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.getBootstrapServers());
+        return AdminClient.create(adminClientConfig);
+    }
+
+    default ConnectRunner newConnectRunner(final KafkaContainer kafka,
+                                                  final File pluginDir,
+                                                  final int offsetFlushIntervalMs) {
+        return new ConnectRunner(pluginDir, kafka.getBootstrapServers(), offsetFlushIntervalMs);
+    }
+
+    default KafkaProducer<byte[], byte[]> newProducer(final KafkaContainer kafka) {
+        final Map<String, Object> producerProps = new HashMap<>();
+        producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.getBootstrapServers());
+        producerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+            "org.apache.kafka.common.serialization.ByteArraySerializer");
+        producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+            "org.apache.kafka.common.serialization.ByteArraySerializer");
+        return new KafkaProducer<>(producerProps);
+    }
+
+    default Future<RecordMetadata> sendMessageAsync(final KafkaProducer<byte[], byte[]> producer,
+                                                    final String topicName,
+                                                    final int partition,
+                                                    final String key,
+                                                    final String value) {
+        final ProducerRecord<byte[], byte[]> msg = new ProducerRecord<>(
+            topicName, partition,
+            key == null ? null : key.getBytes(),
+            value == null ? null : value.getBytes());
+        return producer.send(msg);
+    }
+
+    default Map<String, String> basicConnectorConfig(final String connectorName) {
+        final Map<String, String> config = new HashMap<>();
+        config.put("name", connectorName);
+        config.put("key.converter", "org.apache.kafka.connect.converters.ByteArrayConverter");
+        config.put("value.converter", "org.apache.kafka.connect.converters.ByteArrayConverter");
+        config.put("tasks.max", "1");
+        return config;
+    }
+
+    static void startTestingProcess(File pluginDir) throws IOException, InterruptedException {
```

Comment:
`startTestingProcess` doesn't seem much corresponding to what's going on in the function. Maybe rename it to something like `extractConnectorPlugin` or the like?

Reply:
- rename